ga4gh/backend.py

```diff
@@ -395,8 +403,21 @@ def referenceSetsGenerator(self, request):
         Returns a generator over the (referenceSet, nextPageToken) pairs
         defined by the specified request.
         """
-        return self._topLevelObjectGenerator(
-            request, self.getNumReferenceSets(), self.getReferenceSetByIndex)
+        results = []
+        for obj in self.getReferenceSets():
+            include = True
+            if request.md5checksum is not None:
+                if request.md5checksum != obj.getMd5Checksum():
+                    include = False
+            if request.accession is not None:
```

Comment:
or maybe there's already enough control flow logic here and avoiding a small time penalty isn't worth further cluttering the code...